simple_static_website/main.tf

```
@@ -50,7 +50,7 @@ resource "random_string" "suffix" {
 }
 
 resource "aws_s3_bucket" "this" {
-  bucket = "${var.domain_name_source}-${random_string.suffix.result}"
+  bucket = var.s3_bucket_name == null ? "${var.domain_name_source}-${random_string.suffix.result}" : var.s3_bucket_name
```

This just needs to be tweaked like so:
```suggestion
  bucket = var.s3_bucket_name == "" ? "${var.domain_name_source}-${random_string.suffix.result}" : var.s3_bucket_name
```